File: app/src/main/java/de/tum/in/tumcampus/models/Kino.java
package de.tum.in.tumcampus.models;

import java.util.Date;

import de.tum.in.tumcampus.auxiliary.Utils;

/**
 * Kino Object
 */
public class Kino {

    // all entries in the kino database
    public final String id;
    public final String title;
    public final String year;
    public final String runtime;
    public final String genre;
    public final String director;
    public final String actors;
    public final String rating;
    public final String description;
    public final String cover;
    public final String trailer;
    public final Date date;
    public final Date created;

    /**
     * Kino Constructor
     * @param id
     * @param title
     * @param year
     * @param runtime
     * @param genre
     * @param director
     * @param actors
     * @param rating
     * @param description
     * @param cover
     * @param trailer
     * @param date
     * @param created
     */
    public Kino(String id, String title, String year, String runtime, String genre, String director, String actors,
                String rating, String description, String cover, String trailer, Date date, Date created){
        this.id = id;
        this.title = title;
        this.year = year;
        this.runtime = runtime;
        this.genre = genre;
        this.director = director;
        this.actors = actors;
        this.rating = rating;
        this.description = description;
        this.cover = cover;
        this.trailer = trailer;
        this.date = date;
        this.created = created;
    }

    public String toString(){
        return "id=" + id + " title=" + title + " year=" + year + " runtime=" + runtime +
                " genre=" + genre + " director=" + director + " actors=" + actors + " rating=" + rating +
                " description=" + description + " cover=" + cover + " trailer=" + trailer +
                " date=" + Utils.getDateString(date) + " created=" + Utils.getDateString(created);
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/model/Kino.kt
package de.tum.`in`.tumcampusapp.component.ui.tufilm.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.RoomWarnings
import com.google.gson.annotations.SerializedName
import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormat
import java.util.*
import kotlin.math.roundToInt

/**
 * Kino Constructor
 *
 * @param id          ID
 * @param title       Title
 * @param year        Year
 * @param runtime     Runtime
 * @param genre       Genre
 * @param director    Director
 * @param actors      Actors
 * @param rating      IMDB-Rating
 * @param description Description
 * @param cover       Cover
 * @param trailer     Trailer
 * @param date        Date
 * @param created     Created
 * @param link        Link
 */
@Entity
@SuppressWarnings(RoomWarnings.DEFAULT_CONSTRUCTOR)
data class Kino(
        @PrimaryKey
        @SerializedName("kino")
        val id: String,
        val title: String,
        val year: String,
        val runtime: String,
        val genre: String,
        val director: String,
        val actors: String,
        val rating: String,
        val description: String,
        val cover: String,
        val trailer: String?,
        val date: DateTime,
        val created: DateTime,
        val link: String
) {

    val formattedShortDate: String
        get() {
            // e.g. 11/20/2018 8:00 PM (Style SS = short date and short time)
            val shortDate = DateTimeFormat.forPattern("dd MMM").print(date)
            val shortTime = DateTimeFormat.shortTime().withLocale(Locale.getDefault()).print(date)
            return "$shortDate\n$shortTime"
        }

    val formattedDescription: String
        get() {
            return description
                    .replace(".", ". ")   // Add space after full stops
                    .replace("\\s+", " ") // If this results in multiple spaces, reduce them to one
                    .replace("\n", "")
                    .replace("\r", "\r\n")
                    .removeSuffix("\r\n")
        }

    val trailerSearchUrl: String
        get() {
            val actualTitle = title.split(": ")[1]
            return "https://www.youtube.com/results?search_query=Trailer $actualTitle".replace(" ", "+")
        }

    val formattedRating: String
        get() {
            val roundedRating = rating.toDouble().roundToInt()
            return "$roundedRating / 10"
        }

}
